### PR TITLE
add api call function

### PR DESCRIPTION
**Pull request by baidang201**

### CODE REVIEW

File: stakepad-ui/utils/api.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import protobuf from './protobuf.js'
+import fetch from 'node-fetch';
+import {logger} from './log.js'
+
+const URL = 'http://127.0.0.1:3000'
+export async function probufFetcher (url, data, resMessage) {
```
Comment (amphineko) [inline stakepad-ui/utils/api.js]: typo

Comment (amphineko) [reply stakepad-ui/utils/api.js]: suggest renaming to fetchProtobuf

File: stakepad-ui/utils/api.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import protobuf from './protobuf.js'
+import fetch from 'node-fetch';
+import {logger} from './log.js'
+
+const URL = 'http://127.0.0.1:3000'
```
Comment (amphineko) [inline stakepad-ui/utils/api.js]: suggest using environment variable

**Review by amphineko — CHANGES_REQUESTED** [review]
BTW, you may convert your protoFetcher into ES class. So your `URL` could be passed as a constructor parameter.

File: stakepad-ui/utils/api.js (hunk 1 above)
Comment (baidang201) [reply stakepad-ui/utils/api.js]: done

File: stakepad-ui/utils/api.js (hunk 2 above)
Comment (baidang201) [reply stakepad-ui/utils/api.js]: done

**Pull request merged**